tests/utils_test.py

```diff
@@ -45,6 +49,53 @@ def setUp(self):
     def tearDown(self):
         os.environ = self.os_environ
 
+    def test_convert_volume_binds_empty(self):
+        self.assertEqual(convert_volume_binds({}), [])
+        self.assertEqual(convert_volume_binds([]), [])
+
+    def test_convert_volume_binds_list(self):
+        data = ['/a:/a:ro', '/b:/c:z']
+        self.assertEqual(convert_volume_binds(data), data)
+
+    def test_convert_volume_binds_complete(self):
+        data = {
+            '/mnt/vol1': {
+                'bind': '/data',
+                'mode': 'ro'
+            }
+        }
+        self.assertEqual(convert_volume_binds(data), ['/mnt/vol1:/data:ro'])
+
+    def test_convert_volume_binds_compact(self):
+        data = {
+            '/mnt/vol1': '/data'
+        }
+        self.assertEqual(convert_volume_binds(data), ['/mnt/vol1:/data:rw'])
+
+    def test_convert_volume_binds_no_mode(self):
+        data = {
+            '/mnt/vol1': {
+                'bind': '/data'
+            }
+        }
+        self.assertEqual(convert_volume_binds(data), ['/mnt/vol1:/data:rw'])
+
+    def test_convert_volume_binds_unicode(self):
+        expected = ['/mnt/지연:/unicode/박:rw']
+
+        if six.PY2:
+            expected[0] = expected[0].decode('utf-8')
+
+        data = {
+            '/mnt/지연': {
+                'bind': '/unicode/박',
+                'mode': 'rw'
+            }
+        }
```

Comment:
These will be byte strings in Python 2 - is that correct? Do we care what happens when users pass in `unicode` strings in Python 2?